cdap/resource_application.go

```diff
@@ -154,15 +95,7 @@ func resourceApplicationDelete(d *schema.ResourceData, m interface{}) error {
 func resourceApplicationExists(d *schema.ResourceData, m interface{}) (bool, error) {
 	config := m.(*Config)
 	name := d.Get("name").(string)
-
-	namespace := d.Get("namespace").(string)
-	if exists, err := namespaceExists(config, namespace); err != nil {
-		return false, fmt.Errorf("failed to check for existence of namespace %q: %v", namespace, err)
-	} else if !exists {
-		return false, nil
-	}
-
-	addr := urlJoin(config.host, "/v3/namespaces", namespace, "/apps")
+	addr := urlJoin(config.host, "/v3/namespaces", d.Get("namespace").(string), "/apps")
 	req, err := http.NewRequest(http.MethodGet, addr, nil)
```

Comment:
Do we need to check for 404?

If an error is returned then the exist check is considered to have failed. We will need to return false, nil if its a 404

Reply:
I think 404 should be failed exists check and should return `false, nil`.

We can't return `nil, err` cuz `go vet` will complain:
```
cannot convert nil (untyped nil value) to bool
```

Reply:
Sorry I think I explained wrong... I think for this to work properly it needs to be:

req, err := http.NewRequest(http.MethodGet, addr, nil)
var e *httpError
if errors.As(err, &e) && e.code == 404 {
  return false, nil
} else {
  return false, err
}

if err is not nil the exists value doesn't matter, Terraform will say the exist check failed.

Reply:
Also you can now remove all the code below to unmarshal etc since if the GET works and has no error we can assume it exists.